core/controllers/resources_test.py

```diff
@@ -215,3 +215,181 @@ def test_bad_extensions_are_detected(self):
+        response_dict = self.post_json(
+            '%s/0' % self.AUDIO_UPLOAD_URL_PREFIX,
+            {'filename': 'test.mp3'},
+            csrf_token=csrf_token,
+            expect_errors=True,
+            expected_status_int=400,
+            upload_files=(('audio', 'unused_filename', 'non_audio_data'),)
+        )
+        self.logout()
+        self.assertEqual(response_dict['code'], 400)
+        self.assertEqual(response_dict['error'], 'Audio not recognized as '
+                         'a mp3 file')
+
+    def test_missing_extensions_are_detected(self):
+        """Test upload of filenames with no extensions are caught"""
+
+        self.login(self.EDITOR_EMAIL)
+        response = self.testapp.get('/create/0')
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        missing_extension_filename = 'test'
+        with open(os.path.join(feconf.TESTS_DATA_DIR, self.TEST_AUDIO_FILE),
+                  mode='rb') as f:
+            raw_audio = f.read()
+        response_dict = self.post_json(
+            '%s/0' % (self.AUDIO_UPLOAD_URL_PREFIX),
+            {'filename': missing_extension_filename},
+            csrf_token=csrf_token,
+            expect_errors=True,
+            expected_status_int=400,
+            upload_files=(('audio', 'unused_filename', raw_audio),)
+        )
+        self.logout()
+        self.assertEqual(response_dict['code'], 400)
+        self.assertEqual(response_dict['error'],
+                         'No filename extension: it should have '
+                         'one of the following extensions: '
+                         '%s' % feconf.ACCEPTED_AUDIO_EXTENSIONS.keys())
+
+    def test_exceed_max_length_detected(self):
+        """Test that audio file is less than max playback length"""
+
+        self.login(self.EDITOR_EMAIL)
+        response = self.testapp.get('/create/0')
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        with open(os.path.join(feconf.TESTS_DATA_DIR,
+                               self.TEST_AUDIO_FILE_OVER_MAX_LENGTH),
+                  mode='rb') as f:
+            raw_audio = f.read()
+        response_dict = self.post_json(
+            '%s/0' % (self.AUDIO_UPLOAD_URL_PREFIX),
+            {'filename': 'test.mp3'},
+            csrf_token=csrf_token,
+            expect_errors=True,
+            expected_status_int=400,
+            upload_files=(('audio', 'unused_filename', raw_audio),)
+        )
+        self.logout()
+        self.assertEqual(response_dict['code'], 400)
+        self.assertIn('Audio must be under %s seconds in length'
+                      % feconf.MAX_AUDIO_FILE_LENGTH_SEC,
+                      response_dict['error'])
+
+    # The following test should only be used when more than
+    # one audio extension has been allowed in feconf.ACCEPTED_AUDIO_EXTENSIONS
+    # The flac file filler should be replaced with an accepted format.
+
+    # def test_non_matching_extensions_are_detected(self):
```

Comment:
Okay I uncommented it and just added a check to only run it when there's more than one audio extension, reason being that it checks when an audio file is uploaded with an accepted filename extension, but is actually recognized as a different accepted audio format when reading the raw data.